File: IT_FrontEnd/src/app/model/SupportTicket/support-ticket.ts
import {Status} from "../../enums/status";
import { UserU } from "../UserU/userU";
import {Breakdown} from "../Breakdow/breakdown";
import {Equipment} from "../Equipment/equipment";
import {Technician} from "../Technician/technician";

export class SupportTicket {
  id!: number;
  description!: string;
  createdDate!: Date;
  ticketStatus!: Status;
  userU?: UserU;
  technician?: Technician;
  equipment?: Equipment;
  breakdown?: Breakdown;
}

File: IT_BackEnd/src/main/java/com/support_App/model/SupportTicket.java
package com.support_App.model;


import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.support_App.enums.Status;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class SupportTicket {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String description ;
    private LocalDate createdDate ;
    @Enumerated(EnumType.STRING)
    @Column(name = "ticketStatus" , nullable = false , length = 225)
    private Status ticketStatus;


    @ManyToOne(fetch = FetchType.LAZY)
    @JsonBackReference
    @JsonIgnore
    private UserU userU;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonBackReference
    @JsonIgnore
    private Technician technician;


    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnore
    @JoinColumn(name = "equipment_id")
    private Equipment equipment;


    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnore
    @JoinColumn(name = "breakdown_id")
    private Breakdown breakdown;

}
